boring-sys/build.rs

```diff
@@ -538,41 +538,46 @@ fn build_boring_from_sources() -> String {
 }
 
 fn link_in_precompiled_bcm_o(bssl_dir: &str) {
-    println!("cargo:warning=linking in precompiled `bcm.o` module");
+    const BCM_O_NAME: &str = "bcm.c.o";
+    println!("cargo:warning=linking in precompiled {BCM_O_NAME}` module");
 
-    let bcm_o_src_path = env::var("BORING_SSL_PRECOMPILED_BCM_O")
+    let precompiled_bcm_o_src_path = env::var("BORING_SSL_PRECOMPILED_BCM_O")
         .expect("`fips-link-precompiled` requires `BORING_SSL_PRECOMPILED_BCM_O` env variable to be specified");
 
     let libcrypto_path = PathBuf::from(bssl_dir)
-        .join("build/crypto/libcrypto.a")
+        .join("build/libcrypto.a")
         .canonicalize()
         .unwrap()
         .display()
         .to_string();
 
-    let bcm_o_dst_path = PathBuf::from(bssl_dir).join("build/bcm-fips.o");
```

Comment:
This is the critical line. The "shadow" build adds gives the precompiled module a distinctive name so that `bcm-fips.o` and `bcm.o` are both included in `libcrypto.a`. In this PR, we give the precompiled module _the same name_ so that it _replaces_ `bcm.o`.